docs/safety-eval/safety.md

```diff
@@ -0,0 +1,101 @@
+
+## Running on an interactive session
+
+Run the following command from the main project dir `open-instruct`.
+### Safety benchmarks
+
+For all benchmarks requiring safety evaluation unless noted otherwise, as a default, we use the [WildGuard](https://github.com/allenai/wildguard) classifier to evaluate the safety of model outputs.
+
+- [WildGuardTest](https://arxiv.org/abs/2406.18495)
+- [Harmbench](https://arxiv.org/abs/2402.04249)
+- [ToxiGen](https://arxiv.org/abs/2203.09509)
+- [XSTest](https://arxiv.org/abs/2308.01263)
+- [JailbreakTrigger (in TrustLLM)](https://arxiv.org/abs/2401.05561)
+- [Do-anything-now](https://arxiv.org/abs/2308.03825)
+- [WildJailbreak](https://arxiv.org/abs/2406.18510) (both harmful and benign contrast sets)
+
+```commandline
+PYTHONPATH=safety-eval python safety-eval/evaluation/run_all_generation_benchmarks.py    \
+ --model_name_or_path allenai/tulu-2-dpo-7b     --model_input_template_path_or_name tulu2    \
+  --report_output_path ./generation_results/metrics.json     --save_individual_results_path ./generation_results/all.json \
+  --hf_upload_name {HF upload name} --upload_to_hf {HF repo ID}
+```
+
+**Changing classifiers for safety benchmarks**:
+
+You can change the safety classifier used for evaluation by specifying the `classifier_model_name` in the yaml file.
+For example, when you want to use the HarmBench's classifiers for evaluation on HarmBench, you can use `HarmbenchClassifier` as the `classifier_model_name`. Please check out the `evaluation/tasks/generation/harmbench/default.yaml` and `evaluation/tasks/classification/harmbench/harmbench_classsifier.yaml` to see the classifier's specification.
+
+
+
+
+## Running on a Beaker Batch Job
+
+The recommended way to run evaluation on Beaker is through [beaker-gantry](https://github.com/allenai/beaker-gantry). The basic format of a gantry command is `gantry run {beaker arguments} -- python {script path from repo root} {python script arguments}`. 
+
+**Important**: Before you run any command with gantry, make sure you *commit and push*, since gantry will attempt to clone the repo with your local latest commit hash.
+
+See the "One-Time Setup" section below before running commands. To test your setup, run the following command -- if this job succeeds, then you're ready to run evaluations with gantry.
+
+```bash
+gantry run --workspace {workspace} --budget ai2/oe-adapt --beaker-image kavelr/oe-safety --venv base --cluster ai2/mosaic-cirrascale --env-secret OPENAI_API_KEY=openai_api_key --env-secret HF_TOKEN=hf_token -- python -c 'print("Hello world")'
```

Comment:
is there a way to run without gantry? My guess is with a pre-built image, then we might not even need to clone the repo in. We just setup a beaker config pointing at the right image and gives the right commands (which we have example code for in `submit_eval_jobs`...

Reply:
It should definitely work with `submit_eval_jobs`. I just find it easier to use granty as it requires minimal effort to run.

Reply:
Cool! Let me test that then... gantry is fine, but if we have a prebuilt image with the safety stuff then we can edit the dockerfile to clone the repo there, rather than have it as a submodule here.

Reply:
The image is `seungjuh/oe-safety-support-olmo17`